eng/build.ps1

```diff
@@ -20,46 +20,98 @@ Param(
 
 function Get-Help() {
   Write-Host "Common settings:"
-  Write-Host "  -subset                   Build a subset, print available subsets with -subset help (short: -s)"
-  Write-Host "  -vs                       Open the solution with VS using the locally acquired SDK. Path or solution name (ie -vs Microsoft.CSharp)"
-  Write-Host "  -os                       Build operating system: Windows_NT, Linux, OSX, or Browser"
-  Write-Host "  -arch                     Build platform: x86, x64, arm, arm64, or wasm (short: -a). Pass a comma-separated list to build for multiple architectures."
-  Write-Host "  -configuration            Build configuration: Debug, Release or [CoreCLR]Checked (short: -c). Pass a comma-separated list to build for multiple configurations"
-  Write-Host "  -runtimeConfiguration     Runtime build configuration: Debug, Release or [CoreCLR]Checked (short: -rc)"
-  Write-Host "  -librariesConfiguration   Libraries build configuration: Debug or Release (short: -lc)"
-  Write-Host "  -verbosity                MSBuild verbosity: q[uiet], m[inimal], n[ormal], d[etailed], and diag[nostic] (short: -v)"
-  Write-Host "  -binaryLog                Output binary log (short: -bl)"
-  Write-Host "  -help                     Print help and exit (short: -h)"
+  Write-Host "  -arch (-a)                     Target platform: x86, x64, arm, arm64, or wasm."
+  Write-Host "                                 Pass a comma-separated list to build for multiple architectures."
+  Write-Host "                                 [Default: Your machine's architecture.]"
+  Write-Host "  -binaryLog (-bl)               Output binary log."
+  Write-Host "  -configuration (-c)            Build configuration: Debug, Release or Checked."
+  Write-Host "                                 Checked is exclusive to the CLR subset. Use it for optimized code with asserts and"
```

Comment:
This is incorrect: `Checked` flavor also sets the `DEBUG` macro. The only difference between `Debug` and `Checked` flavors is that the `Checked` flavor is compiled with compiler and linker optimizations, so it runs faster. I would write (here and elsewhere):

```
Checked is exclusive to the CLR subset. It is the same as Debug, but is compiled with optimizations enabled so runs faster.
```

The downside of being compiled with optimizations is it is sometimes harder to debug. But we don't need to write a novel here :-)

Reply:
Oh I see. I had misunderstood how the DEBUG macro is enabled. Thanks for the clarification!

Reply:
Fixed it. I also updated the previous comment with the latest output.